pypit/spectrographs/bpmimage.py

```diff
@@ -0,0 +1,175 @@
+          Set reduce_badpix='bias'
+          Provide msbias image
+       4. Dummy image from shape
+
+    Parameters
+    ----------
+    spectrograph : str (optional)
+       Used to specify properties of the detector (for processing)
+       Attempt to set with settings['run']['spectrograph'] if not input
+    settings : dict (optional)
+      Settings for trace slits
+    det : int, optional
+      Detector index
+      Required for LRISr and DEIMOS
+    binning : Table (optional)
+        Could remove if we put binning in settings['detector']
+    shape : tuple
+      Image shape;  used to construct a dummy BPM if all else fails
+    msbias : ndarray, optional
+      Used to construct the BPM if reduce_badpix=='bias'
+    reduce_badpix : str, optional
+      'bias' -- Build from bias images
+
+    Attributes
+    ----------
+    bpm : ndarray
+
+    """
+    # Keep order same as processimages (or else!)
+    def __init__(self, spectrograph=None, settings=None, det=None,
+                 binning=None, reduce_badpix=None, msbias=None, shape=None):
+
+        # Parameters unique to this Object
+        self.spectrograph = spectrograph
+        self.binning = binning
+        self.det = det
+
+        self.reduce_badpix = reduce_badpix
+        self.msbias = msbias
+
+        self.shape = shape
+        self.settings = settings
+
+        # Checks
+        if (self.reduce_badpix == 'bias') and (self.msbias is None):
+            msgs.error("Need to supply msbias image with this option")
+        if (self.spectrograph == 'keck_deimos') and (self.det is None):
+            msgs.error("Need to supply det with this option")
+        if (self.spectrograph == 'keck_lris_red') and (self.det is None):
+            msgs.error("Need to supply det with this option")
+        if (self.spectrograph == 'keck_lris_red') and (self.binning is None):
+            msgs.error("Need to supply binning with this option")
+
+        # Attributes (set after init)
+        self.frametype = frametype
+
+        # Output
+        self.bpm = None
+
+
+    def build(self):
+        """
+        Generate the BPM Image
+
+        Returns
+        -------
+        self.bpm
+
+        """
+        if self.reduce_badpix == 'bias':
+            # Get all of the bias frames for this science frame
+            if self.msbias is None:
+                msgs.warn("No bias frame provided!")
+                msgs.info("Not preparing a bad pixel mask")
+                return False
+            # TODO -- Deal better with this datasec kludge
+            datasec = []
+            for i in range(self.settings['detector']['numamplifiers']):
+                sdatasec = "datasec{0:02d}".format(i+1)
+                datasec.append(self.settings['detector'][sdatasec])
+            # Construct
+            self.bpm = arprocimg.badpix(self.msbias, self.settings['detector']['numamplifiers'], datasec)
+        else:
+            # Instrument dependent
```

Comment:
Just checking, in case this is an error... didn't we want to include this in each respective Instrument class?

Reply:
Indeed, although I am also capturing these 'errors'
in the Calibrations class now.